docs/dev/compiler-toolchain/overview.md

```diff
@@ -0,0 +1,68 @@
+# Overview
+
+This section introduces an LLVM-based compiler toolchain for smart contract languages with Ethereum Virtual Machine
+(EVM) support. The toolchain works on top of existing compilers and requires their output, which typically includes
+intermediate representations (IRs), abstract syntax trees (ASTs), and auxiliary contract metadata and documentation.
+::: info
+At the time of writing, we support Solidity and Vyper.
+:::
+
+The toolchain consists of the following:
+
+1. [High-level source code compilers](#high-level-source-code-compilers): `solc` and `vyper`.
+2. [IR compilers, or LLVM front-end](#ir-compilers): `zksolc` and `zkvyper`.
+3. [The LLVM framework](./llvm.md) with a zkEVM back-end which emits the zkEVM text assembly.
+4. [The assembler](#assembler) which produces the zkEVM bytecode from the text assembly.
+5. [Hardhat plugins](#hardhat-plugins) which set up the environment.
+
+## High-level Source Code Compilers
+
+High-level source code compilers are original upstream compilers called by IR compilers as
+child processes to process their output. We use two such compilers at the time of writing:
+
+- [solc](https://github.com/ethereum/solc-bin): the official Solidity compiler. For more info, see [the documentation](https://docs.soliditylang.org/en/latest/).
+- [vyper](https://github.com/vyperlang/vyper/releases): the official Vyper compiler. For more info, see [the documentation](https://docs.vyperlang.org/en/latest/index.html).
+
+## IR Compilers
+
+Our toolchain includes LLVM front-ends that process the output of high-level source code compilers:
+
+- [zksolc](https://github.com/matter-labs/zksolc-bin); calling `solc` as a child process. For more info, see [the documentation](./solidity.md).
+- [zkvyper](https://github.com/matter-labs/zkvyper-bin); calling `vyper` as a child process. For more info, see [the documentation](./vyper.md).
+
+These IR compilers perform the following steps:
+
+1. Receive the input, which is usually standard or combined JSON passed by the Hardhat plugin via the standard input.
+2. Modify the input, save the relevant data, and pass the input down to the underlying high-level source code compiler - which is called as a child process.
```

Comment:
The hyphen is unidiomatic here, unless you’re trying for strong and unusual emphasis; I’d suggest just a comma.  (Otherwise it should be changed to a dash rather than a hyphen.)

Reply:
High-level is the hyphen.
The dash is instead of parenthesis as it is a separate statement.

Reply:
Flash is probably saying that that symbol is an actual hyphen, as the dash `—` is not easily printable with some layouts.

Reply:
Both:  The ASCII hyphen for a dash is an antiquated typographic mistake, and a dash here is unidiomatic in English:  either a pair of parentheses or a comma would be correct.  (I’d prefer a comma stylistically.)